### PR TITLE
Add --orientation option to pagerotate command

### PR DESCRIPTION
**Pull request by gatesphere**
Added '--orientation' option to pagerotate command, to allow conditionally rotating a page to a target orientation.

#### Description

Allows conditionally rotating a page with the pagerotate command by specifying a target orientation.  If the rotated page would *not* match the target orientation, pagerotate does nothing.  Defaults to 'any', default behavior is unchanged.

#### Checklist

- [x] feature/fix implemented
- [x] code formatting ok (`black` and `ruff check .`)
- [x] `mypy` returns no error
- [x] tests added/updated and `pytest` succeeds
- [ ] documentation added/updated
    - [x] command docstring and option/argument `help`
    - [ ] README.md updated (Feature Overview)
    - [x] CHANGELOG.md updated
    - [ ] added new command to `reference.rst`
    - [ ] RTD doc updated and building with no error (`make clean && make html` in `docs/`)

### CODE REVIEW

File: vpype_cli/operations.py [hunk 1]
```diff
@@ -662,12 +662,21 @@ def layout(
     is_flag=True,
     help="Rotate clockwise instead of the default counter-clockwise",
 )
+@click.option(
+    "--orientation",
+    "-o",
+    type=click.Choice(["any", "portrait", "landscape"]),
+    default="any",
```
Comment (abey79) [inline vpype_cli/operations.py:669]: - I think we can do without `any`. If we use that option, it means we want a specific orientation.
- Better use `ChoiceType` wrapper, so it supports substitution (`--orientation %some_var%`).

```suggestion
    type=ChoiceType(["portrait", "landscape"]),
```

File: vpype_cli/operations.py [hunk 2]
```diff
@@ -677,6 +686,12 @@ def pagerotate(document: vp.Document, clockwise: bool):
         return document
     w, h = page_size
 
+    # check orientation constraint, and do nothing if target orientation
+    # won't match desired result
+    if (orientation == "portrait" and h > w) or (orientation == "landscape" and w > h):
```
Comment (abey79) [inline vpype_cli/operations.py:691]: A test for `orientation is None` is needed here because of the change above.

File: vpype_cli/operations.py [hunk 3]
```diff
@@ -662,12 +662,21 @@ def layout(
...
+@click.option(
+    "--orientation",
+    "-o",
+    type=click.Choice(["any", "portrait", "landscape"]),
+    default="any",
+    help="Conditionally rotate only if the final orientation matches this option",
+)
 @global_processor
-def pagerotate(document: vp.Document, clockwise: bool):
+def pagerotate(document: vp.Document, clockwise: bool, orientation: str):
```
Comment (abey79) [inline vpype_cli/operations.py:673]:
```suggestion
def pagerotate(document: vp.Document, clockwise: bool, orientation: str | None):
```

**Review by abey79 — CHANGES_REQUESTED** [review]
Very useful addition, thanks! Some minor comments, should be straightforward to fix (mores so than the failing doc build with which I'm currently struggling—unrelated to the present PR).

File: vpype_cli/operations.py (hunk 2 above)
Comment (gatesphere) [reply vpype_cli/operations.py:691]: Is an explicit test needed?  If `orientation == None`, then `orientation == "portrait"` will evaluate False.  The code seems to work fine without an explicit test here.

Comment (abey79) [reply vpype_cli/operations.py:697]: I'd be surprised if `mypy` agreed with that, but let's see :) In any case, I generally dislike such implicit tests for `None`. Also, arguably any other string input should be checked and trigger a `logging.warning()`, and that requires an explicit check for `None`.

Comment (abey79) [reply vpype_cli/operations.py:697]: well apparently mypy is on your side :)

Comment (gatesphere) [reply vpype_cli/operations.py:697]: mypy or not, I'm good either way.  I can add it if you'd like.

Guido's 'explicit is better than implicit' perhaps applies here.

Comment (abey79) [reply vpype_cli/operations.py:697]: The more I think of it, we should have that warning for (non-`None`) garbage input.

Comment (abey79) [reply vpype_cli/operations.py:697]: I mean, if the use says `pagerotate --orientation garbage`. This should trigger a warning.

Comment (gatesphere) [reply vpype_cli/operations.py:697]: I do realize now that we can't rely on click for people using it via the API...

Comment (gatesphere) [reply vpype_cli/operations.py:704]: Added a check to the most recent commit.

**Review by abey79 — APPROVED** [review]
Thanks for bearing with me—great addition!

**Pull request merged**